code/modules/reagents/chemistry/machinery/smoke_machine.dm

```diff
@@ -66,7 +66,7 @@
 
 /obj/machinery/smoke_machine/attackby(obj/item/I, mob/user, params)
 	add_fingerprint(user)
-	if(istype(I, /obj/item/reagent_containers))
+	if(istype(I, /obj/item/reagent_containers) && (I.container_type & OPENCONTAINER_1))
```

Comment:
there's an is_open_container proc or something

Reply:
I'm searching for one and I'm not sure that there is, actually